### ISSUE TITLE
Commits used for dockerfiles

### ISSUE THREAD

**Issue by ksemaev** [vacuumlabs/adalite-backend-service#65]
@refi93 TY very much for making this fork and providing instructions! I used this compose as a template - https://github.com/vacuumlabs/adalite-backend-service/blob/new-importer/docker/docker-compose.yml but for some reason when I try to access backend service on `/api/v2/txs/signed'` I get ```Error broadcasting tx: { code: 'Internal', message: 'Error trying to connect with importer' }``` while the backend itself sends this to log: ```Error trying to connect with importer (req.route=/api/v2/txs/signed, res.status=500, res.time=1688, res.restCode=Internal)
    POST /api/v2/txs/signed HTTP/1.1```
Indeed there's no `v2` path on importer, but as I see from code - backend should proxy requests to importer removing that `v2` part. But it doesn't for some reason.

Because of security restrictions I had to build images from the repos myself, so maybe I'm using wrong commit? I used a1eadb1900b34eedbda03d80623b445c0e0fed18 for backend and ae12acd2efc4c2b0e8739727a9174c0193bf21b4 for importer. Should I maybe use another ones?

Please advice!

**Comment by refi93** [comment]
The url to the importer tx sending endpoint is set by the `IMPORTER_ENDPOINT` environment variable in the docker file: https://github.com/vacuumlabs/adalite-backend-service/blob/new-importer/docker/docker-compose.yml#L25 and here is the respective config file transforming it to `sendTxEndpoint`: https://github.com/vacuumlabs/adalite-backend-service/blob/new-importer/config/custom-environment-variables.json#L15

I'm not sure why it would fail, but is the `adalite-importer` container actually running and pushing data to the database?

**Comment by ksemaev** [comment]
yep, I have this `IMPORTER_ENDPOINT=http://adalite-importer:8080/api/txs/signed`, and some of requests indeed work. Importer is up and running, health is OK. 

For example I can access my adalite importer like that:
```
curl http://adalite-importer:8080/api/txs/signed
{"code":"MethodNotAllowed","message":"GET is not allowed"}
curl http://adalite-importerr:8080/api/v2/txs/signed
{"code":"ResourceNotFound","message":"/api/v2/txs/signed does not exist"}
```

When I access backend with plain curl I get the same:
```
curl https://adalite-backend/api/v2/txs/signed
{"code":"MethodNotAllowed","message":"GET is not allowed"}
```

But when I try some POST like 
`curl -v -XPOST -H "Content-type: application/json" -d '{"signedTx": "goOfggDYGFgkglgg2Tl3BhKUDg6zKIIWI+x1O34Xys9IbPJmu7AHGSuvxEcB/5+CgtgYWCGDWBwHNAyvvtWg3bXkwfymazombeQGg8yXufAx+TWxoAAaCPoPZBoA5OHAgoLYGFghg1gcY/59dgG6oZVRaSi5EQF1KAFuVOQPY2Caeai6RqAAGsfEN7waB8IlMv+ggYIA2BhYhYJYQG38zu1tPK50hLmnFCZKRhXq2ZRdAXkKTOyG4mgExl9Jdy/jPE1bSAqAK2WDe0IuiyY+djSzINPuINFY+Jqfs+RYQMs2pCUGHDGTUkCl+/wdmvnQXC49yjnUCuue+KUM2FsQ5FQ/tuX8xs2O3qc5alBDrljLYQw3Sv+df/pB5sa8Www="}' 'https://adalite-backend/api/v2/txs/signed'`
I get
```{"code":"Internal","message":"Error trying to connect with importer"}* Closing connection 0```

During last request^ the logs of backend are:
> [2019-12-17T18:59:30.755Z] ERROR: [object Object]/31 on adalite-backend-6665c856f-6rckc: Error trying to connect with importer (req.route=/api/v2/txs/signed, res.status=500, res.time=1407, res.restCode=Internal)
>     POST /api/v2/txs/signed HTTP/1.1

**Comment by ksemaev** [comment]
Don't know if that's useful, but those are my request to importer directly:
```
curl -v -XPOST -H "Content-type: application/json" -d '{"signedTx": "goOfggDYGFgkglgg2Tl3BhKUDg6zKIIWI+x1O34Xys9IbPJmu7AHGSuvxEcB/5+CgtgYWCGDWBwHNAyvvtWg3bXkwfymazombeQGg8yXufAx+TWxoAAaCPoPZBoA5OHAgoLYGFghg1gcY/59dgG6oZVRaSi5EQF1KAFuVOQPY2Caeai6RqAAGsfEN7waB8IlMv+ggYIA2BhYhYJYQG38zu1tPK50hLmnFCZKRhXq2ZRdAXkKTOyG4mgExl9Jdy/jPE1bSAqAK2WDe0IuiyY+djSzINPuINFY+Jqfs+RYQMs2pCUGHDGTUkCl+/wdmvnQXC49yjnUCuue+KUM2FsQ5FQ/tuX8xs2O3qc5alBDrljLYQw3Sv+df/pB5sa8Www="}' 'http://adalite-importer:8080/api/v2/txs/signed'
{"code":"ResourceNotFound","message":"/api/v2/txs/signed does not exist"}
```

And without v2:
```
curl -v -XPOST -H "Content-type: application/json" -d '{"signegYWCGDWBwHNAyvvtWg3bXkwfymazombeQGg8yXufAx+TWxoAAaCPoPZBoA5OHAgoLYGFghg1gcY/59dgG6oZVRaSi5EQF1KAFuVOQPY2Caeai6RqAAGsfEN7waB8IlMv+ggYIA2BhYhYJYQG38zu1tPK50hLmnFCZKRhXq2ZRdAXkKTOyG4mgExl9Jdy/jPE1bSAqAK2WDe0IuiyY+djSzINPuINFY+Jqfs+RYQMs2pCUGHDGTUkCl+/wdmvnQXC49yjnUCuue+KUM2FsQ5FQ/tuX8xs2O3qc5alBDrljLYQw3Sv+df/pB5sa8Www="}' 'http://adalite-importer:8080/api/txs/signed'
"Transaction failed protocol: TransactionRejected"
```

So it looks like backend is not proxying to importer and I can't figure out why....

**Comment by refi93** [comment]
Looking at 

```
curl -v -XPOST -H "Content-type: application/json" -d '{"signegYWCGDWBwHNAyvvtWg3bXkwfymazombeQGg8yXufAx+TWxoAAaCPoPZBoA5OHAgoLYGFghg1gcY/59dgG6oZVRaSi5EQF1KAFuVOQPY2Caeai6RqAAGsfEN7waB8IlMv+ggYIA2BhYhYJYQG38zu1tPK50hLmnFCZKRhXq2ZRdAXkKTOyG4mgExl9Jdy/jPE1bSAqAK2WDe0IuiyY+djSzINPuINFY+Jqfs+RYQMs2pCUGHDGTUkCl+/wdmvnQXC49yjnUCuue+KUM2FsQ5FQ/tuX8xs2O3qc5alBDrljLYQw3Sv+df/pB5sa8Www="}' 'http://adalite-importer:8080/api/txs/signed'
"Transaction failed protocol: TransactionRejected"
```

I think the problem is that either the transaction was submitted already in the past, so it was rejected by the node as the unspent outputs are no longer available, or it is invalid in a different way, but if that's the case, the backend service returning an error 500 is not a correct behavior, of course.

But I still don't completely understand why would the `adalite-backend` report error 500 in that case - my guess though is that the new importer is returning an error status code that causes the "sendTx" call to crash: https://github.com/vacuumlabs/adalite-backend-service/blob/new-importer/src/routes.js#L140 I will look into that today or tomorrow, meanwhile you can check what status code are you getting from the importer when it says "Transaction rejected". You may also try to assemble a valid transaction, try to submit it via the `v2` endpoint and see if it works - my guess is that it should.

To make it clear, for `adalite-backend` container you should be calling the `/api/v2/txs/signed` endpoint and for `adalite-importer` it should be `/api/txs/signed` so it's expected that other variations don't work.

**Comment by ksemaev** [comment]
TY for additional info! I just checked the full response from my request, and it is code 400:
```
root@bash-579b4b7857-9nxvd:/# curl -v -XPOST -H "Content-type: application/json" -d '{"signedTx": "goOfggDYGFgkglgg2Tl3BhKUDg6zKIIWI+x1O34Xys9IbPJmu7AHGSuvxEcB/5+CgtgYWCGDWBwHNAyvvtWg3bXkwfymazombeQGg8yXufAx+TWxoAAaCPoPZBoA5OHAgoLYGFghg1gcY/59dgG6oZVRaSi5EQF1KAFuVOQPY2Caeai6RqAAGsfEN7waB8IlMv+ggYIA2BhYhYJYQG38zu1tPK50hLmnFCZKRhXq2ZRdAXkKTOyG4mgExl9Jdy/jPE1bSAqAK2WDe0IuiyY+djSzINPuINFY+Jqfs+RYQMs2pCUGHDGTUkCl+/wdmvnQXC49yjnUCuue+KUM2FsQ5FQ/tuX8xs2O3qc5alBDrljLYQw3Sv+df/pB5sa8Www="}' 'http://adalite-importer:8080/api/txs/signed'
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying 172.20.224.76...
* TCP_NODELAY set
* Connected to adalite-importer (172.20.224.76) port 8080 (#0)
> POST /api/txs/signed HTTP/1.1
> Host: adalite-importer
> User-Agent: curl/7.58.0
> Accept: */*
> Content-type: application/json
> Content-Length: 400
>
* upload completely sent off: 400 out of 400 bytes
< HTTP/1.1 400 Bad Request
< Server: restify
< Content-Type: application/json
< Content-Length: 50
< Date: Wed, 18 Dec 2019 09:31:29 GMT
< Connection: keep-alive
<
* Connection #0 to host adalite-importer left intact
"Transaction failed protocol: TransactionRejected"
```

So does it mean that backend reacts with `{"code":"Internal","message":"Error trying to connect with importer"}` when it gets 400 from importer?

**Comment by refi93** [comment]
it's possible, I will check. If that's the case I will fix the bug or fill free to submit a PR with the fix yourself

**Comment by ksemaev** [comment]
@refi93 can you please explain one more thing - can I run importer when the code is synced on 50% etc? I mean can it be started at any point of time, or it has to be started only with the `http-bridge` when it started sync from 0?

**Comment by refi93** [comment]
as far as I understand, the http-bridge runs independently from the importer, the importer pulls data from the http-bridge and pushes it to Postgres at its own pace, so the importer can (and inevitably will) be behind the http bridge. Therefore you don't have to reset the bridge state if you reset the importer. I'm not sure though what happens if it's the other way around, but the http-bridge is usually the one that takes longer to sync anyway.

**Comment by roccomuso** [comment]
Was this fixed with #67 ?

**Comment by refi93** [comment]
@roccomuso yes, thanks for pointing out, closing

the error 400 was fixed by #67

**Comment by roccomuso** [comment]
I still got 500 error after the patch.
Did you have a chance to test it?
